streams/src/test/java/org/apache/kafka/streams/integration/KStreamSerDesIntegrationTest.java

```diff
@@ -0,0 +1,491 @@
+        assertTrue(valueTestSerde.configured());
+        validateReceivedMessages(
+                new StringDeserializer(),
+                new StringDeserializer(),
+                Arrays.asList(
+                        KeyValue.pair("A", "A:A"),
+                        KeyValue.pair("B", "B:B"),
+                        KeyValue.pair("C", "C:C"),
+                        KeyValue.pair("D", "D:D"),
+                        KeyValue.pair("E", "E:E")));
+    }
+
+    @Test
+    public void shouldReduceWindowed() throws Exception {
+        final long firstBatchTimestamp = System.currentTimeMillis() - 1000;
+        produceMessages(firstBatchTimestamp);
+        final long secondBatchTimestamp = System.currentTimeMillis();
+        produceMessages(secondBatchTimestamp);
+        produceMessages(secondBatchTimestamp);
+
+        groupedStream
+            .windowedBy(TimeWindows.of(ofMillis(500L)))
+            .reduce(reducer, Materialized.as("reduce-time-windows"))
+            .toStream((windowedKey, value) -> windowedKey.key() + "@" + windowedKey.window().start())
+            .to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+
+        startStreams();
+
+        final long firstBatchWindow = firstBatchTimestamp / 500 * 500;
+        final long secondBatchWindow = secondBatchTimestamp / 500 * 500;
+
+        validateReceivedMessages(
+                new StringDeserializer(),
+                new StringDeserializer(),
+                Arrays.asList(
+                        new KeyValue<>("A@" + firstBatchWindow, "A"),
+                        new KeyValue<>("A@" + secondBatchWindow, "A:A"),
+                        new KeyValue<>("B@" + firstBatchWindow, "B"),
+                        new KeyValue<>("B@" + secondBatchWindow, "B:B"),
+                        new KeyValue<>("C@" + firstBatchWindow, "C"),
+                        new KeyValue<>("C@" + secondBatchWindow, "C:C"),
+                        new KeyValue<>("D@" + firstBatchWindow, "D"),
+                        new KeyValue<>("D@" + secondBatchWindow, "D:D"),
+                        new KeyValue<>("E@" + firstBatchWindow, "E"),
+                        new KeyValue<>("E@" + secondBatchWindow, "E:E")
+                )
+        );
+    }
+
+    static class MyIntSerializer extends IntegerSerializer {
+        boolean configured = false;
+        boolean called = false;
+
+        MyIntSerializer() {
+            super();
+        }
+
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            super.configure(configs, isKey);
+            configured = true;
+        }
+
+        @Override
+        public byte[] serialize(final String topic, final Integer data) {
+            called = true;
+            return super.serialize(topic, data);
+        }
+
+        boolean configured() {
+            return !called || configured;
+        }
+    }
+    static class MyIntDeserializer extends IntegerDeserializer {
```

Comment:
Can we update the test to only use one data type? Don't think we gain any test coverage using multiple -- this seems to be distraction what the purpose of the test is.